src/main/kotlin/ar/edu/unahur/obj2/servidorWeb/Analizador.kt

```diff
@@ -0,0 +1,77 @@
+package ar.edu.unahur.obj2.servidorWeb
+
+import java.time.LocalDateTime
+
+abstract class Analizador() {
+    abstract fun analizar(respuesta: Respuesta)
+}
+
+object analizadorDeRespuestas : Analizador(){
+
+    val demoraMinima = 50
+    val respuestasDemoradas = mutableListOf<Respuesta>()
+
+    override fun analizar(respuesta: Respuesta) {
+        if (respuesta.tiempo > this.demoraMinima){
+            respuestasDemoradas.add(respuesta)
+        }
+    }
+
+    fun respuestasDemoradasModulo(modulo: Modulo): Int {
+        return respuestasDemoradas.filter { respuesta -> respuesta.body == modulo.body }.size
```

Comment:
Lo que tenías que mirar acá es que el módulo que se usó para responder fuera el que viene por parámetro. Lo que hiciste se mete con los detalles de implementación, porque asume que el body de la respuesta es el que configura el módulo. Vamos a hablar sobre esto en la clase de hoy.

Reply:
Otra forma de hacerlo:
```suggestion
        return respuestasDemoradas.count { respuesta -> respuesta.body == modulo.body }
```